solver/src/naive_solver/multi_order_solver.rs

```diff
@@ -43,9 +43,11 @@ pub fn solve(
     let mut orders: Vec<LimitOrder> = orders.collect();
     while !orders.is_empty() {
         let (context_a, context_b) = split_into_contexts(orders.clone().into_iter(), pool);
-        let solution = solve_orders(orders.clone().into_iter(), &pool, &context_a, &context_b)?;
-        if is_valid_solution(&solution) {
-            return Some(solution);
+        if let Some(valid_solution) =
```

Comment:
This change is somewhat unrelated, but it seems wrong to return if the current set of orders isn't solvable (note the ? before). Instead we should use the same strategy as with invalid solutions (remove order with worst limit price and try again).